pkg/db/table.go

```diff
@@ -0,0 +1,288 @@
+// Copyright 2016 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package pkgdb
+
+import (
+	"database/sql"
+	"strings"
+
+	"github.com/juju/errors"
+	"github.com/pingcap/tidb/ast"
+	"github.com/pingcap/tidb/infoschema"
+	"github.com/pingcap/tidb/model"
+	"github.com/pingcap/tidb/mysql"
+	"github.com/pingcap/tidb/parser"
+	"github.com/pingcap/tidb/table"
+	"github.com/pingcap/tidb/types"
+)
+
+// GetSchemaTableWithRowID returns table information with _tidb_rowid column if useRowID is true
+func GetSchemaTableWithRowID(db *sql.DB, schemaName string, tableName string, useRowID bool) (*model.TableInfo, error) {
+	table, err := GetSchemaTable(db, schemaName, tableName)
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+
+	if useRowID && !table.PKIsHandle {
+		addImplicitColumn(table)
+	}
+
+	return table, nil
+}
+
+// GetSchemaTable returns table information.
+func GetSchemaTable(db *sql.DB, schemaName string, tableName string) (*model.TableInfo, error) {
+	createTable, err := GetCreateTable(db, schemaName, tableName)
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+
+	return GetSchemaTableBySQL(createTable, schemaName, tableName)
+}
+
+// GetSchemaTableBySQL returns table information by create table sql.
+func GetSchemaTableBySQL(createTableSQL string, schemaName string, tableName string) (table *model.TableInfo, err error) {
+	stmt, err := parser.New().ParseOneStmt(createTableSQL, "", "")
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+
+	switch s := stmt.(type) {
```

Comment:
If we can ensure it must be a create table statement, we do it with type assert instead of  switch.

`s, ok := stmt.(*ast.CreateTableStmt)`